extractors-canopycover/drone_canopycover.py

```diff
@@ -0,0 +1,715 @@
+    # an image type
+    @property
+    def known_non_image_ext(self):
+        """Returns an array of file extensions that we will see that
+           are definitely not an image type
+        """
+        return ["dbf", "json", "prj", "shp", "shx", "txt"]
+
+    # Look through the file list to find the files we need
+    # pylint: disable=too-many-locals,too-many-nested-blocks
+    def find_image_files(self, files):
+        """Finds files that are needed for extracting plots from an orthomosaic
+
+        Args:
+            files(list): the list of file to look through and access
+
+        Returns:
+            Returns a dict of georeferenced image files (indexed by filename and containing an
+            object with the calculated image bounds as an ogr polygon and a list of the
+            bounds as a tuple)
+
+            The bounds are assumed to be rectilinear with the upper-left corner directly
+            pulled from the file and the lower-right corner calculated based upon the geometry
+            information stored in the file.
+
+            The polygon points start at the upper left corner and proceed clockwise around the
+            boundary. The returned polygon is closed: the first and last point are the same.
+
+            The bounds tuple contains the min and max Y point values, followed by the min and
+            max X point values.
+        """
+        imagefiles = {}
+
+        for onefile in files:
+            ext = os.path.splitext(os.path.basename(onefile))[1].lstrip('.')
+            if not ext in self.known_non_image_ext:
+                if file_is_image_type(self.args.identify_binary, onefile,
+                                      onefile + self.file_infodata_file_ending):
+                    # If the file has a geo shape we store it for clipping
+                    bounds = image_get_geobounds(onefile)
+                    epsg = get_epsg(onefile)
+                    if bounds[0] != np.nan:
+                        ring = ogr.Geometry(ogr.wkbLinearRing)
+                        ring.AddPoint(bounds[2], bounds[1])     # Upper left
+                        ring.AddPoint(bounds[3], bounds[1])     # Upper right
+                        ring.AddPoint(bounds[3], bounds[0])     # lower right
+                        ring.AddPoint(bounds[2], bounds[0])     # lower left
+                        ring.AddPoint(bounds[2], bounds[1])     # Closing the polygon
+
+                        poly = ogr.Geometry(ogr.wkbPolygon)
+                        poly.AddGeometry(ring)
+
+                        ref_sys = osr.SpatialReference()
+                        if ref_sys.ImportFromEPSG(int(epsg)) != ogr.OGRERR_NONE:
+                            logging.error("Failed to import EPSG %s for image file %s",
+                                          str(epsg), onefile)
+                        else:
+                            poly.AssignSpatialReference(ref_sys)
+
```

Comment:
else tell people / logger that this is not a georeferenced image?

Reply:
Added logging